global/matlab/mixtureModel.m

```diff
@@ -0,0 +1,92 @@
+function mixtureModel(inFile,outFile,wbcmd,melocmd)
+% mixtureModel(inFile,outFile,wbcmd,melocmd)
+% Performs Gaussian mixture modeling on precomputed ICs
+% Wrapper arround melodic.Accepts nifti or cifti inputs. 
+% Require Input:
+%   inFile  : file path to IC z-scores, file path including extension as string (accepts nifti or cifti files)
+% Optional Inputs
+%   outFile :  file path to IC z-scores with Gaussian mixture modeling, defaults to inFile 
+%              Can output nifti, nifit-gz, or cifti, depending on extension: .nii, .nii.gz, .dscalar.nii,
+%              Output format does not have to match input format, unless output is cifti.
+%   wbcmd   : worbench command, defaults to 'wb_command' 
+%   melocmd : melodic command, defaults to 'melodic'
+%
+% See also: 
+% https://www.jiscmail.ac.uk/cgi-bin/webadmin?A2=fsl;6e85d498.1607
+% https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/MELODIC#Using_melodic_for_just_doing_mixture-modelling
+% https://www.fmrib.ox.ac.uk/datasets/techrep/tr02cb1/tr02cb1.pdf
+
+% Created by Burke Rosen
+% 2024-07-08
+% Dependencies:workbench, FSL
+% Written with workbench 2.0 and FSL 6.0.7.1
+%
+% ToDo: 
+% Feed arbitrary melodic arguments. 
+% Return workbench errors
+% Save melodic log?
+
+%% handle inputs 
+if nargin < 2 || isempty(outFile); outFile = inFile; end 
```

Comment:
I don't like defaults that destroy their own input data files.  I think outFile should be required.

Reply:
Made required. Now returns help if nargin <2